sweetest/src/main/java/com/mysugr/sweetest/framework/context/DependenciesTestContext.kt

```diff
@@ -83,7 +99,46 @@ class DependenciesTestContext {
      *  [provide]. [provide] automatically requires that the automatically created instance is used.
      */
     fun provide(clazz: KClass<*>) {
-        val dependency = getDependencyConfiguration(clazz)
-        TestEnvironment.dependencies.states[dependency].mode = DependencyMode.REAL
+        val mode = DependencyMode.REAL
+        addDependency(clazz) { state, _ ->
+            state.mode = mode
+        }
+    }
+
+    /**
+     * Checks if the type [clazz] is already added via the global module configuration.
+     * If so, the found [DependencyConfiguration] is added to the [DependencyManager.states].
+     * If not, a [DependencyConfiguration] is created dynamically and also added.
+     */
+    private fun addDependency(clazz: KClass<*>, block: (DependencyState<out Any>, DependencyMode?) -> Unit) {
+        with(TestEnvironment.dependencies) {
+            assertDependency(clazz)
+            val dependencyState = getDependencyConfiguration(clazz)?.let { configuration ->
+                states[configuration]
+            } ?: run {
+                states[DependencyConfiguration(clazz, null, null, null)]
+            }
+
+            block(dependencyState, dependencyState.configuration.defaultDependencyMode)
+        }
+    }
+
+    /**
+     * Asserts if the type [clazz] is already added.
+     */
+    private fun assertDependency(clazz: KClass<*>) {
+        check(TestEnvironment.dependencies.states[clazz] == null) {
+            "Can't change dependency mode or ${clazz.simpleName}, it " +
+                "has already been set before and can't be changed afterwards"
```

Comment:
i don't understand the "has already been set before" part. the check checks whether the state is null. so it can't be have been set already, or maybe i miss something?

Reply:
Hm you are right. That makes no sense. When I think about it, the whole check is redundant as we should just ignore multiple dependency adds.